ClassAttendance/src/ConsoleUI/Contollers/Implementations/SubControllers/StudentController.cs

```diff
@@ -0,0 +1,115 @@
+    {
+        private readonly IStudentService _studentService;
+        private readonly IGroupService _groupService;
+        private readonly StudentMenuView _studentMenu;
+
+        private bool exitFlag;
+
+        public StudentController(IStudentService studentService, StudentMenuView menuView)
+        {
+            _studentService = studentService;
+            _studentMenu = menuView;
+        }
+
+        public async Task HandleInput()
+        {
+            while (!exitFlag)
+            {
+                var input = Console.ReadKey().Key;
+                PrintOperations();
+
+                switch (input)
+                {
+                    case ConsoleKey.D1:
+                        await PrintStudents();
+                        break;
+                    case ConsoleKey.D2:
+                        await PrintStudent();
+                        break;
+                    case ConsoleKey.D3:
+                        await AddStudent();
+                        break;
+                    case ConsoleKey.D4:
+                        await DeleteStudent();
+                        break;
+                    case ConsoleKey.D5:
+                        await UpdateStudent();
+                        break;
+                    case ConsoleKey.D0:
+                        exitFlag = true;
+                        break;
+                    default:
+                        break;
+                }
+            }
+        }
+
+
+        public void PrintOperations()
+        {
+            _studentMenu.PrintMenu();
+        }
+
+        private async Task AddStudent()
+        {
+            var student = _studentMenu.GetPersonFromInput();
+
+            await _studentService.AddAsync(student);
+        }
+
+        private async Task DeleteStudent()
+        {
+            var id = _studentMenu.GetIdFromInput();
+
+            await _studentService.DeleteAsync(id);
+        }
+        
+        private async Task UpdateStudent()
+        {
+            var id = _studentMenu.GetIdFromInput();
+            var student = await _studentService.GetByIdAsync(id);
+            var updatedUser = _studentMenu.UpdateStudent(student);
+
+            await _studentService.UpdateAsync(updatedUser);
+        }
+
+        private async Task PrintStudents()
+        {
+            var students = _studentService.GetAll();
+            var viewModels = new List<StudentViewModel>();
+
+            await foreach (var student in students)
+            {
+                var group = await _groupService.GetGroupByStudentIdAsync(student.Id);
+                viewModels.Add(new StudentViewModel
+                {
+                    Student = student,
+                    GroupName = group.Name,
+                });
+            }
+
+            _studentMenu.PrintStudents(viewModels);
+        }
+
+        private async Task PrintStudent()
+        {
+            var input = Console.ReadLine();
```

Comment:
It's a repeatable part.

Reply:
https://github.com/VladimirZaitsev/VladimirZaitsev_dotnet/pull/4/files#diff-a77f4baac3d674d04d80b06360d71cd0R73

Reply:
I'm not understanding what should i fix here.

Reply:
You already have a method to get I'd from Console. But here you make the same thing again without the using already created method.

Reply:
Fixed